File: js逆向/day14-JS基本功（4）/day14/xialuo01.py

import requests

# 获取JS文件


# 1 执行访问
res  = requests.get('http://127.0.0.1:5001/')
print(res.text)

# 2 传参数
params = {
    'pwd':'123456'
}
res1  = requests.get('http://127.0.0.1:5001/api',params=params)
print(res1.text)

File: js逆向/day14-JS基本功（4）/day14/xialuo01.js
const express = require('express')
const app = express()
const port = 5001  // 程序运行的端口


function ps(){
    return '我是逆向的加密结果！'
}

function pn(val){
    // 环境差异
    pwd = btoa(val)
    return pwd
}


app.get('/', (req, res) => {
    var data = ps()
  res.send(data)
})


app.get('/api', (req, res) => {
    var params = req.query
        console.log(params);
    // 这里是取值
    info = pn(params.pwd)

    res.send(info)
})


app.listen(port, () => {
  console.log(`Example app listening on port ${port}`)
})










